rct229/rulesets/ashrae9012019/ruleset_functions/compare_swh_dist_systems_and_components.py

```diff
@@ -0,0 +1,133 @@
+    )
+    assert_(
+        rmd2_swh_system_and_equip_dict[swh_distribution_id],
+        f"SWH Distribution System {swh_distribution_id} not found in the `rmd2`.",
+    )
+    rmd1_swh_equipment_dict = rmd1_swh_system_and_equip_dict[swh_distribution_id]
+    rmd1_swh_distribution = find_exactly_one_service_water_heating_distribution_system(
+        rmd1, swh_distribution_id
+    )
+    rmd2_swh_equipment_dict = rmd2_swh_system_and_equip_dict[swh_distribution_id]
+    rmd2_swh_distribution = find_exactly_one_service_water_heating_distribution_system(
+        rmd2, swh_distribution_id
+    )
+
+    index_content_dict, compare_context_dict = (
+        (rmd1_swh_distribution, rmd2_swh_distribution)
+        if len(rmd1_swh_distribution) > len(rmd2_swh_distribution)
+        else (rmd2_swh_distribution, rmd1_swh_distribution)
+    )
+
+    compare_context_pair(
+        index_context=index_content_dict,
+        compare_context=compare_context_dict,
+        element_json_path="$.service_water_heating_distribution_systems[*]",
+        extra_schema=EXTRA_SCHEMA["ServiceWaterHeatingDistributionSystem"][
+            "Data Elements"
+        ],
+        error_msg_list=errors_list,
+        search_key=compare_context_str,
+        required_equal=True,
+    )
+
+    # service_water_heating_equipment
+    assert_(
+        len(rmd1_swh_equipment_dict.swh_heating_eq)
+        == len(rmd2_swh_equipment_dict.swh_heating_eq),
+        f"Unequal numbers of SWH Equipment between the two models for {swh_distribution_id}",
+    )
+
+    for swh_eq_id in rmd1_swh_equipment_dict.swh_heating_eq:
+        swh_eq_1 = find_exactly_one_service_water_heating_equipment(rmd1, swh_eq_id)
+        swh_eq_2 = find_exactly_one_service_water_heating_equipment(rmd2, swh_eq_id)
+
+        index_content_dict, compare_context_dict = (
+            (swh_eq_1, swh_eq_2)
+            if len(swh_eq_1) > len(swh_eq_2)
+            else (swh_eq_2, swh_eq_1)
+        )
+
+        compare_context_pair(
+            index_context=index_content_dict,
+            compare_context=compare_context_dict,
+            element_json_path="$.service_water_heating_equipment[*]",
+            extra_schema=EXTRA_SCHEMA["ServiceWaterHeatingEquipment"]["Data Elements"],
+            error_msg_list=errors_list,
+            search_key=compare_context_str,
+            required_equal=True,
+        )
+
+    # pumps
+    assert_(
+        len(rmd1_swh_equipment_dict.pumps) == len(rmd2_swh_equipment_dict.pumps),
+        f"Unequal number of pumps in the two models for SWH Distribution System {swh_distribution_id}",
+    )
+
+    for pump_id in rmd1_swh_equipment_dict.pumps:
+        pump_1 = find_exactly_one_pump(rmd1, pump_id)
+        pump_2 = find_exactly_one_pump(rmd2, pump_id)
+
+        index_content_dict, compare_context_dict = (
+            (pump_1, pump_2) if len(pump_1) > len(pump_2) else (pump_2, pump_1)
+        )
```

Comment:
This is not necessary, following the instruction above.

Reply:
Addressed.